### ISSUE TITLE
Cannot edit any configuration option with unsupported network config

### ISSUE THREAD

**Issue by lathiat** [canonical/lxd-ui#422]
Hi Team,

I have LXD VMs created using the MAAS KVM Pod functionality. It has 3 network interfaces, bridged.

When I try edit the instance configuration in LXD-UI and attempt to change the CPU core count, it fails to save with the error:

> Instance update failed
> Invalid devices: Device validation failed for "eth0": Failed loading device "eth0": Unsupported device type

Despite having not edited that panel. It seems it tries to persist all config options even if not edited.

If I switch to the Networks panel, the network names (eth0/eth1/eth2) show but each "network device" dropdown has a list of all the interfaces on my system (including vsw0/vsw1, which are just linux bridges), but nothing is selected (hence the error, the text shows "Select option). Despite vsw0 being in the list.

Here is the network config from "lxc config show". In case it matters reproduction wise, this VM is in a non-default 'maas' project. 
I guess this bug is in two parts
- Firstly ideally it would not try to persist unchanged configuration particularly if it may have some unsupported feature (there are many ways to configure networks that don't seem currently supported in the UI). 
- Secondly I guess this basic bridged network configuration should work

Versions:
```
LXD snap:
tracking:     latest/stable/ubuntu-22.04
installed:          5.15-002fa0f             (25112) 181MB -

MAAS: 3.3.4
```

```
devices:
  eth0:
    boot.priority: "1"
    name: eth0
    nictype: bridged
    parent: vsw0
    type: nic
  eth1:
    name: eth1
    nictype: bridged
    parent: vsw1
    type: nic
  eth2:
    name: eth2
    nictype: bridged
    parent: vsw1
    type: nic
```

**Comment by edlerd** [comment]
Thank you for reporting. This is a bug, the UI currently does not handle custom network configurations for instances correctly. We'll have it fixed shortly.

In the meantime, you can use the YAML editor to do any changes to those VMs, which will not alter the custom networks unless you edit the YAML code for them.

**Comment by edlerd** [comment]
Fix landed in main next LXD 5.17 should contain it.
